### PR TITLE
Add missing documentation to serde/serde

### PR DESCRIPTION
**Pull request by kakirastern**
To follow up on #2510. 

Aims
====
1. First add missing documentation to serde/serde code base. 
2. Add details to serde/serde code base to make in-code documentation complete.

### CODE REVIEW

**Comment by kakirastern** [comment]
PR is ready for a review.

File: syft/serde/serde.py [hunk 1]
```diff
@@ -114,6 +114,13 @@
 
 ## SECTION: High Level Simplification Router
 def _force_full_simplify(obj: object) -> object:
+    """
+    To force a full simplify genrally if the usual _simplify is not suitable.
+    
+    Args:
+    :param: the object
```
Comment (jvmancuso) [inline syft/serde/serde.py]: Hey, can you please switch this to be a [Google style docstring](http://google.github.io/styleguide/pyguide.html#383-functions-and-methods)? :)

File: syft/serde/serde.py [hunk 2]
```diff
@@ -131,7 +138,15 @@ def _force_full_simplify(obj: object) -> object:
...
 def _generate_simplifiers_and_detailers():
-    """Generate simplifiers, forced full simplifiers and detailers."""
+    """
+    Generate simplifiers, forced full simplifiers and detailers,
+    by registering native and torch types, syft objects with custom
+    simplify and detail methods, or syft objects with custom
+    force_simplify and force_detail methods.
+    
+    Args:
+    :return: the simplifiers, forced_full_simplifiers, detailers
```
Comment (jvmancuso) [inline syft/serde/serde.py]: Same request as above!

File: syft/serde/serde.py [hunk 3]
```diff
@@ -96,10 +96,10 @@
     messaging.SearchMessage,
 ]
 
-# If a object implements its own force_simplify and force_detail functions it should be stored in this list
+# If an object implements its own force_simplify and force_detail functions it should be stored in this list
 OBJ_FORCE_FULL_SIMPLIFIER_AND_DETAILERS = [VirtualWorker]
 
-
+# For egistering syft objects with custom simplify and detail methods
```
Comment (jvmancuso) [inline syft/serde/serde.py]: Typo `egistering -> registering`

File: syft/serde/serde.py (hunk 1 above)
Comment (kakirastern) [reply syft/serde/serde.py]: Sure

**Comment by kakirastern** [comment]
Changes made according to reviewer's suggestions.

File: syft/serde/serde.py [hunk 4]
```diff
@@ -123,6 +123,12 @@
 
 ## SECTION: High Level Simplification Router
 def _force_full_simplify(obj: object) -> object:
+    """To force a full simplify genrally if the usual _simplify is not suitable.
+    
+    Args:
+    :param: the object
```
Comment (mari-linhares) [inline syft/serde/serde.py]: `:param:` should be `param:`. And the same applies to all other arguments and returns.

File: syft/serde/serde.py [hunk 5]
```diff
@@ -123,6 +123,12 @@
...
+    :return: the result from either _simplify or _forced_full_simplify on the object
```
Comment (mari-linhares) [inline syft/serde/serde.py]: Docstring return should be presented as:

```
Returns:
     The result from either _simplify or _forced_full_simplify on the object.
```

Here's a nice docstring example: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

**Comment by kakirastern** [comment]
Changes made according to reviewer's suggestions

File: syft/serde/serde.py [hunk 6]
```diff
@@ -140,7 +148,16 @@ def _force_full_simplify(obj: object) -> object:
 
 ## SECTION: dinamically generate simplifiers and detailers
 def _generate_simplifiers_and_detailers():
-    """Generate simplifiers, forced full simplifiers and detailers."""
+    """Generate simplifiers, forced full simplifiers and detailers,
+    by registering native and torch types, syft objects with custom
+    simplify and detail methods, or syft objects with custom
+    force_simplify and force_detail methods.
+    
+    Args: None
```
Comment (mari-linhares) [inline syft/serde/serde.py]: No need to have args section if there are no args.

File: syft/serde/serde.py [hunk 7]
```diff
@@ -325,8 +343,10 @@ def apply_zstd_compression(decompressed_input_bin) -> tuple:
     Apply ZSTD compression to the input
 
     Args:
-        :param decompressed_input_bin: the binary to be compressed
-        :return: a tuple (compressed_result, ZSTD)
+        param decompressed_input_bin: the binary to be compressed
```
Comment (mari-linhares) [inline syft/serde/serde.py]: param -> param:

**Comment by kakirastern** [comment]
PR is ready for another review.

File: syft/serde/serde.py [hunk 8]
```diff
@@ -123,6 +123,14 @@
 
 ## SECTION: High Level Simplification Router
 def _force_full_simplify(obj: object) -> object:
+    """To force a full simplify genrally if the usual _simplify is not suitable.
+    
+    Args:
+        param: the object
```
Comment (robert-wagner) [inline syft/serde/serde.py]: The arg here is obj rather than param

File: syft/serde/serde.py [hunk 9]
```diff
@@ -123,6 +123,14 @@
...
+    
+    Returns:
+        the result from either _simplify or _forced_full_simplify on the object
```
Comment (robert-wagner) [inline syft/serde/serde.py]: Nit. The simplified object

File: syft/serde/serde.py [hunk 10]
```diff
@@ -314,8 +328,10 @@ def apply_lz4_compression(decompressed_input_bin) -> tuple:
     Apply LZ4 compression to the input
 
     Args:
-        :param decompressed_input_bin: the binary to be compressed
-        :return: a tuple (compressed_result, LZ4)
+        param: decompressed_input_bin: the binary to be compressed
```
Comment (robert-wagner) [inline syft/serde/serde.py]: Param isn't needed for google style doc strings

File: syft/serde/serde.py [hunk 11]
```diff
@@ -325,8 +341,10 @@ def apply_zstd_compression(decompressed_input_bin) -> tuple:
     Apply ZSTD compression to the input
 
     Args:
-        :param decompressed_input_bin: the binary to be compressed
-        :return: a tuple (compressed_result, ZSTD)
+        param: decompressed_input_bin: the binary to be compressed
```
Comment (robert-wagner) [inline syft/serde/serde.py]: ditto

File: syft/serde/serde.py [hunk 12]
```diff
@@ -123,6 +123,14 @@
 
 ## SECTION: High Level Simplification Router
 def _force_full_simplify(obj: object) -> object:
+    """To force a full simplify genrally if the usual _simplify is not suitable.
+    
+    Args:
+        The simplified object
```
Comment (robert-wagner) [inline syft/serde/serde.py]: I was saying the simpified object is the return not the arg

Comment (kakirastern) [reply syft/serde/serde.py]: Sorry

**Pull request merged**